### DISCUSSION TITLE
Deploy in Groups

### DISCUSSION THREAD

**Discussion by sesantanajr** [svrooij/WinTuner#124] [General]
I deployed the 7zip app to test it by applying it to groups, but I couldn't. There was no error, it just doesn't show assigned to a mandatory, registered group. Anyway, I couldn't get it to work. The parameters I used were the ones below, is that correct?

New-WtWingetPackage -PackageId 7zip.7zip -PackageFolder C:\Wintuner\Apps | Deploy-WtWin32App -RequiredFor "GS - Suporte TI" -UninstallFor "GS - RH"

New-WtWingetPackage -PackageId 7zip.7zip -PackageFolder C:\Wintuner\Apps | Deploy-WtWin32App -RequiredFor "GS - Suporte TI" -UninstallFor "GS - RH"
INFO: Packaging package 7zip.7zip 24.08
INFO: [WingetManager] Getting package info for 7zip.7zip 24.08 from github
INFO: [IntuneManager] Generating IntuneWin package for 7zip.7zip 24.08 X64 System in C:\Wintuner\Apps
INFO: [IntuneManager] Downloading installer from https://7-zip.org/a/7z2408-x64.msi to C:\Users\sesan\AppData\Local\Temp\wintuner\7zip.7zip\24.08\7z2408-x64.msi
INFO: [Packager] Creating Intune package from C:\Users\sesan\AppData\Local\Temp\wintuner\7zip.7zip\24.08
INFO: [Packager] Creating package for C:\Users\sesan\AppData\Local\Temp\wintuner\7zip.7zip\24.08\7z2408-x64.msi in C:\Users\sesan\AppData\Local\Temp\wintuner\7zip.7zip\24.08 to C:\Wintuner\Apps\7zip.7zip\24.08
INFO: [Packager] Compressing the source folder C:\Users\sesan\AppData\Local\Temp\wintuner\7zip.7zip\24.08 to C:\Users\sesan\AppData\Local\Temp\447d22c3-e8d7-455c-9364-fefc07522349\IntuneWinPackage\Contents\IntunePackage.intunewin
INFO: [Packager] Generating application info
INFO: [Packager] Encrypting file C:\Users\sesan\AppData\Local\Temp\447d22c3-e8d7-455c-9364-fefc07522349\IntuneWinPackage\Contents\IntunePackage.intunewin
INFO: [Packager] Generated detection XML file C:\Users\sesan\AppData\Local\Temp\447d22c3-e8d7-455c-9364-fefc07522349\IntuneWinPackage\Metadata\Detection.xml
INFO: [Packager] Done creating package for C:\Users\sesan\AppData\Local\Temp\wintuner\7zip.7zip\24.08\7z2408-x64.msi in C:\Users\sesan\AppData\Local\Temp\wintuner\7zip.7zip\24.08 to C:\Wintuner\Apps\7zip.7zip\24.08
INFO: [IntuneManager] Downloading logo from https://api.winstall.app/icons/7zip.7zip.png
INFO: [DefaultFileManager] Skipping download of https://api.winstall.app/icons/7zip.7zip.png to C:\Wintuner\Apps\7zip.7zip\logo.png because the file already exists
INFO: [IntuneManager] Writing detection info with msi details 7zip.7zip {23170F69-40C1-2702-2408-000001000000}
INFO: [IntuneManager] Writing package readme for package 7zip.7zip
INFO: Loading package details from folder C:\Wintuner\Apps\7zip.7zip\24.08
INFO: Uploading Win32App 7-Zip to Intune with file C:\Wintuner\Apps\7zip.7zip\24.08\7z2408-x64.intunewin
INFO: [GraphAppUploader] Creating new Win32LobApp
INFO: [GraphAppUploader] Creating new content version for app bebe7626-4be1-40b2-b383-00c15ae1ca1e
INFO: [ChunkedAzureFileUploader] Uploading C:\Users\sesan\AppData\Local\Temp\c75f7c89-beea-46ae-b723-95decda2d369\IntuneWinPackage\Contents\IntunePackage.intunewin to https://mmcswda01.blob.core.windows.net/ea9dd78d-ef10-410d-83f8-1f901d39220b/467da70f-c6e2-4bf1-9e5a-e571948c8f3a/d30fc21b-117e-4b4d-9ae3-74bc62f7b398.intunewin.bin?sv=2017-04-17&sr=b&si=-1982974357&sig=AU12JtKwdfm8F4fhubu%2FVTbGUGemjl68dk%2FeHzjmCPw%3D
INFO: [GraphAppUploader] Added content version 1 to app bebe7626-4be1-40b2-b383-00c15ae1ca1e
INFO: Created Win32App 7-Zip with id bebe7626-4be1-40b2-b383-00c15ae1ca1e
INFO: Assigning app bebe7626-4be1-40b2-b383-00c15ae1ca1e to groups

AllowAvailableUninstall         : True
ApplicableArchitectures         : X64
DetectionRules                  : {Microsoft.Graph.Beta.Models.Win32LobAppProductCodeDetection}
DisplayVersion                  : 24.08
InstallCommandLine              : msiexec /i 7z2408-x64.msi /qn /norestart
InstallExperience               : Microsoft.Graph.Beta.Models.Win32LobAppInstallExperience
MinimumCpuSpeedInMHz            :
MinimumFreeDiskSpaceInMB        :
MinimumMemoryInMB               :
MinimumNumberOfProcessors       :
MinimumSupportedOperatingSystem : Microsoft.Graph.Beta.Models.WindowsMinimumOperatingSystem
MinimumSupportedWindowsRelease  : 2004
MsiInformation                  : Microsoft.Graph.Beta.Models.Win32LobAppMsiInformation
RequirementRules                : {}
ReturnCodes                     : {Microsoft.Graph.Beta.Models.Win32LobAppReturnCode,
                                  Microsoft.Graph.Beta.Models.Win32LobAppReturnCode,
                                  Microsoft.Graph.Beta.Models.Win32LobAppReturnCode,
                                  Microsoft.Graph.Beta.Models.Win32LobAppReturnCode…}
Rules                           : {Microsoft.Graph.Beta.Models.Win32LobAppProductCodeRule}
SetupFilePath                   : 7z2408-x64.msi
UninstallCommandLine            : msiexec /x {23170F69-40C1-2702-2408-000001000000} /qn /norestart
CommittedContentVersion         : 1
ContentVersions                 :
FileName                        : 7z2408-x64.intunewin
Size                            : 1924432
Assignments                     :
Categories                      :
CreatedDateTime                 : 28/09/2024 21:31:16 +00:00
DependentAppCount               : 0
Description                     : 7-Zip is a file archiver with a high compression ratio.
Developer                       : Igor Pavlov
DisplayName                     : 7-Zip
InformationUrl                  : https://7-zip.org/
IsAssigned                      : False
IsFeatured                      : False
LargeIcon                       : Microsoft.Graph.Beta.Models.MimeContent
LastModifiedDateTime            : 28/09/2024 21:31:16 +00:00
Notes                           : Generated by WingetIntune at 28/09/2024 21:30:50 +00:00 [WinTuner|Winget|7zip.7zip]
Owner                           :
PrivacyInformationUrl           :
Publisher                       : Igor Pavlov
PublishingState                 : Published
Relationships                   :
RoleScopeTagIds                 : {0}
SupersededAppCount              : 0
SupersedingAppCount             : 0
UploadState                     : 1
AdditionalData                  : {[@odata.context,
                                  https://graph.microsoft.com/beta/$metadata#deviceAppManagement/mobileApps/$entity]}
BackingStore                    : Microsoft.Kiota.Abstractions.Store.InMemoryBackingStore
Id                              : bebe7626-4be1-40b2-b383-00c15ae1ca1e
OdataType                       : #microsoft.graph.win32LobApp

**Comment by svrooij** [comment]
Have you checked the docs? 

https://wintuner.app/docs/wintuner-powershell/Deploy-WtWin32App

**Reply by sesantanajr** [reply]
Yes, it doesn't even show an error when executing the command.
Even using the -RequiredFor AvailableFor -UninstallFor parameters with the Group name in quotation marks, it didn't work. He does not associate with any group.
Now if I use AllDevices or AllUsers it works normally.

**Reply by svrooij** [reply]
The docs say

> Groups that the app should available for, **Group Object ID** or AllUsers / AllDevices

Can you help us with a better explanation, that you can only use `AllUsers` or `AllDevives` or the **object ID** of the group? There is no support for display names.

Reason: Intune needs the object id, to get the id from a name we would have to do a lookup (which requires an extra scope).

And it does not give an error, because it filters all other values out. Would you like it to crash?
